app/signer/bls.go

```diff
@@ -223,60 +292,93 @@ func (k *BlsKey) BlsPubKey() (bls12381.PublicKey, error) {
 
 // LoadBlsSignerIfExists attempts to load an existing BLS signer from the specified home directory
 // Returns the signer if files exist and can be loaded, or nil if files don't exist
-// If customKeyPath is provided, will use that path for the BLS key file instead of the default.
-func LoadBlsSignerIfExists(homeDir string, customKeyPath string) checkpointingtypes.BlsSigner {
-	blsKeyFile := customKeyPath
-	if blsKeyFile == defaultBlsKeyFilePath || blsKeyFile == "" {
-		blsKeyFile = DefaultBlsKeyFile(homeDir)
+// Password precedence:
+// 1. Environment variable (BABYLON_BLS_PASSWORD)
+// 2. Custom password file path (if provided)
+// 3. Default password file path
+func LoadBlsSignerIfExists(homeDir string, customPasswordPath, customKeyPath string) checkpointingtypes.BlsSigner {
+	blsKeyFile := determineKeyFilePath(homeDir, customKeyPath)
+	if !cmtos.FileExists(blsKeyFile) {
+		return nil
 	}
-	blsPasswordFile := DefaultBlsPasswordFile(homeDir)
 
-	if !cmtos.FileExists(blsKeyFile) || !cmtos.FileExists(blsPasswordFile) {
-		return nil
+	bls := LoadBls(blsKeyFile, customPasswordPath)
+	if bls != nil {
+		return &bls.Key
+	}
+
+	defaultPasswordFile := DefaultBlsPasswordFile(homeDir)
+	if cmtos.FileExists(defaultPasswordFile) {
+		if bls, err := LoadBlsFromFile(blsKeyFile, defaultPasswordFile); err == nil {
+			return &bls.Key
+		}
```

Comment:
This bit is not clear to me. Can we decide the password path first and then call `LoadBls` once?